src/core/messages.ts

```diff
@@ -277,6 +293,130 @@ export interface Messages extends EmitsDiscontinuities {
   get channel(): Ably.RealtimeChannel;
 }
 
+interface MessageReactionEvent {
+  type: 'message-reaction-create' | 'message-reaction-remove';
+  refSerial: string;
+  refType: string;
+  reaction: string;
+  clientId: string;
+  timestamp: Date;
+}
+
+export type MessageReactionListener = (event: MessageReactionSummaryEvent) => void;
+export type MessageRawReactionListener = (event: MessageReactionEvent) => void;
+
+export interface Unsubscribable {
+  unsubscribe: () => void;
+}
+
+interface Reactions {
+  add(message: { serial: string }, refType: ReactionRefType, reaction: string, count?: number): Promise<void>;
+  remove(message: { serial: string }, refType: ReactionRefType, reaction: string): Promise<void>;
+  subscribe(listener: MessageReactionListener): Unsubscribable;
+  subscribeRaw(listener: MessageRawReactionListener): Unsubscribable;
+}
+
+export class DefaultMessageReactions implements Reactions {
+  private _emitter = new EventEmitter<{ reaction: MessageReactionEvent; summary: MessageReactionSummaryEvent }>();
+
+  constructor(
+    private readonly _logger: Logger,
+    private readonly _options: MessageOptions | undefined,
+    private readonly _api: ChatApi,
+    private readonly _roomID: string,
+    private readonly _channel: Ably.RealtimeChannel,
+  ) {
+    void _channel.subscribe(this._processMessageEvent.bind(this));
+    if (this._options?.RawMessageReactions) {
+      void _channel.annotations.subscribe(this._processAnnotationEvent.bind(this));
+    }
+  }
+
+  private _processAnnotationEvent(event: Ably.Annotation) {
+    console.log('ably-chat-js: annotation received:', event);
+  }
+
+  private _processMessageEvent(event: Ably.InboundMessage) {
+    // only process summary events
+    if (event.action !== ChatMessageActions.MessageAnnotationSummary) {
+      return;
+    }
+    // they must have a refSerial
+    if (!event.refSerial) {
+      return;
```

Comment:
Worth a `warn` log here, as this is unintended behaviour?

Reply:
added debug logs. I don't think they should be `warn` since it's nothing wrong that this user did, they just received something that's not complete, so will be ignored.

Seems we do _debug_ in DefaultMessages `_processEvent` as well.